hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -629,7 +629,6 @@ public boolean delete(Data key) {
 
         final Record record = getRecordOrNull(key, now, false);
         if (record == null) {
-            removeIndex(record);
```

Comment:
@jerrinot you seems right but maybe there was another intention. @ahmetmircik can you have a look?